azure-pipelines.yml

```diff
@@ -138,7 +138,7 @@ jobs:
   displayName: 'Verify src/azure-cli/requirements.*.Darwin.txt'
   condition: succeeded()
   pool:
-    vmImage: 'macOS-12'
+    vmImage: ${{ variables.macos_pool }}
```

Comment:
> vmImage string.
Name of the VM image you want to use; valid only in the Microsoft-hosted pool.

Ref: https://learn.microsoft.com/en-us/azure/devops/pipelines/yaml-schema/pool?view=azure-pipelines#pool-name-demands-vmimage